utils/R/convert.input.R

```diff
@@ -6,65 +6,144 @@
+  print(paste(
+    "start CHECK Convert.Inputs", fcn, input.id, host$name, outfolder, formatname, mimetype, 
+    site.id, start_date, end_date))
+  existing.dbfile <- dbfile.input.check(
+    siteid=site.id, mimetype=mimetype, formatname=formatname, parentid=input.id, 
+    con=con, hostname=host$name, ignore.dates=TRUE)
+  print(existing.dbfile, digits=10)
   print("end CHECK")
-  print(check)
-  if(length(check)>0){
-    return(list(input.id=check$container_id, dbfile.id=check$id))
+
+  if(nrow(existing.dbfile) > 0) {
+    if(nrow(existing.dbfile) > 1) {
+      print(existing.dbfile)
+      logger.warn("Multiple existing inputs found. Using last.")
+      existing.dbfile <- existing.dbfile[nrow(existing.dbfile),]
+    }
+
+    existing.input <- db.query(paste0("SELECT * FROM inputs WHERE id=", existing.dbfile[['container_id']]), con)    
+    # Convert dates to Date objects and strip all time zones (DB values are timezone-free)
+    start_date <- force_tz(as_date(start_date), 'GMT')
+    end_date <- force_tz(as_date(end_date), 'GMT')
+    existing.input$start_date <- force_tz(as_date(existing.input$start_date), 'GMT')
+    existing.input$end_date <- force_tz(as_date(existing.input$end_date), 'GMT')
+
+    if(overwrite) {
+      # start and end dates stay the same. 
+      # collect files to flag for deletion
+      existing.dbfile <- dbfile.check('Input', existing.input[['id']], con, host$name)
+      if(nrow(existing.dbfile) > 0) {
+        # There should only be one existing dbfile record, but by passing all paths we will get all files anyway
+        files.to.delete <- remote.execute.R(paste0(
+          "list.files('", existing.dbfile[['file_path']], "', full.names=TRUE)"), 
+          host, user=NA, verbose=TRUE, R="R")
+
+        file.deletion.commands <- .get.file.deletion.commands(files.to.delete)
+        
+        remote.execute.R(file.deletion.commands$move.to.tmp, host, user=NA, verbose=TRUE, R="R")
+        
+        # Schedule files to be replaced or deleted on exiting the function
+        succesful <- FALSE
+        on.exit(
+          if(successful) {
+            logger.info("Conversion successful, with overwrite=TRUE. Deleting old files.")
+            remote.execute.R(file.deletion.commands$delete.tmp, host, user=NA, verbose=TRUE, R="R")
+          } else {
+            logger.info("Conversion failed. Replacing old files.")
+            remote.execute.R(file.deletion.commands$replace.from.tmp, host, user=NA, verbose=TRUE, R="R")
+          }
+        )
+      }
+    } else if((start_date >= existing.input$start_date) && (end_date <= existing.input$end_date)) {
+      # There's an existing input that spans desired start/end dates. Use that one. 
+      logger.info("Skipping this input conversion because files are already available.")
+      return(list(input.id=existing.input$id, dbfile.id=existing.dbfile$id))
+    } else {
+      # Start/end dates need to be updated so that the input spans a continuous timeframe
```

Comment:
If I understood correctly, this is the desired behavior: If there's any matching input record already (minus dates), then it should be extended to accommodate its entire current range and the newly requested start/end range. Anyway, that's what this version does. Sucks to be you if 2000-2001 exists and now you want to run 2015-2016, because you have to wait for all intervening years to complete...